src/gui/src/InstrumentEditor/InstrumentEditor.cpp

```diff
@@ -1290,6 +1295,61 @@ void InstrumentEditor::update()
 	}
 }
 
+int InstrumentEditor::getPointSize() const {
```

Comment:
There seem to be quite a few instances of `getPointSize` and `on_PreferencesChanged` with almost identical code. Would it be worth considering adding some common superclass that encapsulates that functionality? Like
```c++

template < int s, int n, int l >
class WidgetThatHasAFont {
protected:
  int getPointSize() {
    switch Preferences::FontSize::Small:  return s;
  [ ... etc ]
};

class InstrumentEditor : public QWidget, protected WidgetThatHasAFont< 10, 12, 14 > {
...
};
```
?

Reply:
Ah, nice. I wasn't aware templates can be used in such an elegant way.

Reply:
On reflection, I must have been tired -- I have no idea why I seem to have started writing a `switch` and then skipped straight to the `case` value... :D

Reply:
Haha, Don't worry. I also jumped over it and didn't noticed.

All the copy&pasting for the `onPreferencesChanged`  did bug me too. But in the end it's content should contain class-specific assignments to member variables and updates, like `update()`, `createBackground()`, or the update of a label - and in the future the update of everything else that can be changed in the `Preferences` class through a session.

Maybe something like a  `std::vector<std::pair<memberVariable,functionPointerToPreferencesGetter*>` registered in the constructor + a custom overwrite implementations for classes requiring additional stuff like rerendering would do the trick. But the member variables are of different types and a `std::variant` is only available starting from C++17 while we AFAIK still work with C++14. And since C++ does not feature reflection, storing a string containing the name of the member variable wouldn't do any good either. So, I just sticked to the classical version